### PR TITLE
Fix #1046

### LINKED ISSUE #1046: Sleeping in a bed causes error in console
### Issue description

Sleeping in a bed caused an error in console.
Tested in Creative mode.

[11:21:26] [Server thread/CRITICAL]: TypeError: "Argument 1 passed to pocketmine\utils\BinaryStream::putBool() must be of the type boolean, null given, called in phar://C:/Users/Mark/Desktop/1.1/PocketMine-MP.phar/src/pocketmine/network/mcpe/protocol/SetSpawnPositionPacket.php on line 63" (EXCEPTION) in "/src/pocketmine/utils/BinaryStream__64bit" at line 92

### Steps to reproduce the issue

1. Wait for night
2. Sleep in bed

### OS and versions

* PocketMine-MP:  PocketMine-MP 1.6.2dev-168
* PHP: 7
* Server OS: Win 10
* Game Version: PE/Win10 (delete as appropriate)

### Plugins
- Test on a clean server without plugins: is the issue reproducible without any plugins loaded?
Yes

If the issue is **not** reproducible without plugins:
- Have you asked for help on our forums before creating an issue?
- Can you provide sample, *minimal* reproducing code for the issue? If so, paste it in the bottom section
- Paste your list of plugins here (use the 'plugins' command in PocketMine-MP)

### Crashdump, backtrace or other files

* N/A

### PR DESCRIPTION
**Pull request by Misteboss**
Fix #1046 issue, it was happening because spawnForced field wasn't filled

### CODE REVIEW

File: src/pocketmine/Player.php [hunk 1]
```diff
@@ -1184,6 +1184,7 @@ public function setSpawn(Vector3 $pos){
 		$pk->y = (int) $this->spawnPosition->y;
 		$pk->z = (int) $this->spawnPosition->z;
 		$pk->spawnType = SetSpawnPositionPacket::TYPE_PLAYER_SPAWN;
+		$pk->spawnForced = true;
```
Comment (dktapps) [inline src/pocketmine/Player.php]: I'm not sure this should be true... spawnForced means players always spawn at their spawn point when they join the server.

Other than that, looks good.

**Comment by dktapps** [comment]
Don't worry about Travis failing... there's something wrong with PECL so all the builds are crashing.

File: src/pocketmine/Player.php (hunk 1 above)
Comment (Misteboss) [inline src/pocketmine/Player.php]: Idk cause it's a PlayerSpawn and not a World Spawn so.. ?

Comment (dktapps) [inline src/pocketmine/Player.php]: I'd leave it as false anyway, vanilla always sends false for this field.

Comment (dktapps) [inline src/pocketmine/Player.php]: This pull request is primarily fixing a core bug, feature additions should be done separately.

Comment (Misteboss) [inline src/pocketmine/Player.php]: Ok no problem ^^

**Pull request merged**